### PR TITLE
LINK_CONTRACT_ADDRESS as chain specific config

### PR DESCRIPTION
**Pull request by vyzaldysanchez**

### CODE REVIEW

**Comment by vyzaldysanchez** [comment]
I think this would require us to update the Chain related UI to support GQL. I don't think that has happened yet. @jkongie

File: core/web/evm_chains_controller_test.go [hunk 1]
```diff
@@ -10,16 +10,18 @@ import (
...
+	"gopkg.in/guregu/null.v4"
+
 	"github.com/smartcontractkit/chainlink/core/chains/evm/types"
 	"github.com/smartcontractkit/chainlink/core/internal/testutils/evmtest"
 	"github.com/smartcontractkit/chainlink/core/utils"
 	"github.com/smartcontractkit/chainlink/core/web/presenters"
-	"github.com/stretchr/testify/assert"
-	"gopkg.in/guregu/null.v4"
+
+	"github.com/stretchr/testify/require"
```
Comment (pinebit) [inline core/web/evm_chains_controller_test.go:21]: I think this needs to be sorted/grouped.

File: core/chains/evm/types/types.go [hunk 2]
```diff
@@ -76,6 +76,7 @@ type ChainCfg struct {
 	MinRequiredOutgoingConfirmations               null.Int
 	MinimumContractPayment                         *assets.Link
 	OCRObservationTimeout                          *models.Duration
+	LinkContractAddress                            null.String
```
Comment (samsondav) [inline core/chains/evm/types/types.go:79]: Can we sort these alphabetically while you're here

File: core/web/evm_chains_controller_test.go [hunk 3]
```diff
@@ -79,6 +81,7 @@ func Test_EVMChainsController_Show(t *testing.T) {
 					BlockHistoryEstimatorBlockHistorySize: null.IntFrom(50),
 					EvmEIP1559DynamicFees:                 null.BoolFrom(true),
 					MinIncomingConfirmations:              null.IntFrom(12),
+					LinkContractAddress:                   null.StringFrom(utils.ZeroAddress.String()),
```
Comment (samsondav) [inline core/web/evm_chains_controller_test.go:84]: Let's use a random address here to make sure we get it right `cltest.NewAddress` outta do it

File: core/web/resolver/evm_chain_test.go [hunk 4]
```diff
@@ -361,6 +366,7 @@ func TestResolver_CreateChain(t *testing.T) {
 					EthTxReaperThreshold:            &threshold,
 					GasEstimatorMode:                null.StringFrom("BlockHistory"),
 					ChainType:                       null.StringFrom("optimism"),
+					LinkContractAddress:             null.StringFrom(utils.ZeroAddress.String()),
```
Comment (samsondav) [inline core/web/resolver/evm_chain_test.go:369]: Again here lets randomly generate one

**Review by samsondav — CHANGES_REQUESTED** [review]

File: core/web/evm_chains_controller_test.go (hunk 3 above)
Comment (vyzaldysanchez) [reply core/web/evm_chains_controller_test.go:84]: Understood!

File: core/web/resolver/evm_chain_test.go (hunk 4 above)
Comment (vyzaldysanchez) [reply core/web/resolver/evm_chain_test.go:369]: Understood!

File: core/chains/evm/types/types.go (hunk 2 above)
Comment (vyzaldysanchez) [reply core/chains/evm/types/types.go:79]: Of course

File: core/chains/evm/config/config_test.go [hunk 5]
```diff
@@ -98,6 +98,27 @@ func TestChainScopedConfig(t *testing.T) {
...
+		t.Run("uses chain-specific override value when that is set", func(t *testing.T) {
+			val := "0x5431F5F973781809D18643b87B44921b11355d81"
+			evmconfig.UpdatePersistedCfg(cfg, func(cfg *evmtypes.ChainCfg) {
+				cfg.LinkContractAddress = null.StringFrom(val)
+			})
+
+			assert.Equal(t, val, cfg.LinkContractAddress())
+		})
+		t.Run("uses global value when that is set", func(t *testing.T) {
+			val := "0x5431F5F973781809D18643b87B44921b11355d81"
```
Comment (prashantkumar1982) [inline core/chains/evm/config/config_test.go:116]: Also set chain specific override here to a different value.
Then verify that the one used was infact the global value, not this chain specific value.

**Review by prashantkumar1982 — APPROVED** [review]

**Review by samsondav — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**